demos/social_distance_demo/cpp/include/person_trackers.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+#include <unordered_map>
+#include <tuple>
+#include <deque>
+
+#include <opencv2/core.hpp>
+
+class TrackableObject {
+public:
+    TrackableObject(cv::Rect2i bb, const std::vector<float> &r, cv::Point centroid)
+            : bbox{bb}, reid{r}, updated{false}, disappeared(0) {
+        centroids.push_back(centroid);
+    }
+
+    cv::Rect bbox;
+    std::vector<float> reid;
+    std::vector<cv::Point> centroids;
+    bool updated;
+    int disappeared;
+};
+
+class PersonTrackers {
+public:
+    PersonTrackers() : trackIdGenerator{0}, similarityThreshold{0.7f}, maxDisappeared{10} {}
+
+    void similarity(std::list<TrackableObject> &tos) {
+        if (trackables.size() > 0) {
+            for (const auto& to : tos) {
+                std::deque<std::pair<int, float>> sim;
+                for (auto &tracker : trackables) {
+                    if (!tracker.second.updated) {
+                        float cosine = cosineSimilarity(to.reid, tracker.second.reid);
+                        if (cosine > similarityThreshold) {
+                            sim.push_back(std::make_pair(tracker.first, cosine));
+                        }
+                    }
+                }
+
+                if (!sim.empty()) {
+                    auto maxSimilarity = std::max_element(sim.begin(), sim.end(), [](std::pair<int, float> a, std::pair<int, float> b) {
```

Comment:
```suggestion
                    int maxSimilarity = std::max_element(sim.begin(), sim.end(), [](std::pair<int, float> a, std::pair<int, float> b) {
```
`int` is shoreter and more explicit.

Reply:
Ok, done.